enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,498 @@
+
+Day-2:
+- Managed Services: TBD: Patch CIO.
+
+### Workflow Description
+
+#### OpenShift Self-managed
+
+- User:
+  - Creates `install-config.yaml` enabling the use of Security Group, example `platform.aws.ingressController.securityGroupEnabled`, **when** `lbType=NLB` (already exists).
+```yaml
+# install-config.yaml
+platform:
+  aws:
+    region: us-east-1
+    lbType: NLB                   <-- deprecate by platform.aws.ingressController.loadBalancerType?
+    ingressController:            <-- proposing to aggregate CIO configurations
+      securityGroupEnabled: True  <-- new field
+[...]
+```
+- Installer:
+  - The installer will generate the CIO manifests, setting the `managedSecurityGroup` field in the `networkLoadBalancer` section of the IngressController `spec`:
+```yaml
+# $INSTALL_DIR/manifests/cluster-ingress-default-ingresscontroller.yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  creationTimestamp: null
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+[...]
+  endpointPublishingStrategy:
+    loadBalancer:
+      providerParameters:
+        aws:
+          networkLoadBalancer:
+            managedSecurityGroup: true   <-- new field
+          type: NLB
+        type: AWS
+      scope: External
+    type: LoadBalancerService
+[...]
+```
+- Cluster Ingress Operator (CIO):
+  - CIO will create the Service instance for the default router. If the `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.managedSecurityGroup` is `true`, CIO will include the following annotation in the Service manifest:
+```yaml
+# Manifest for Service XYZ is created with annotations:
+apiVersion: v1
+kind: Service
+metadata:
+  name: echoserver
+  namespace: mrbraga
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: nlb
+    service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+    service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true" <-- new annotation
+[...]
+```
+- Cloud Controller Manager (CCM):
+  - CCM validates the annotation to manage SG on NLB, creates the SG and rules, and pass the SG ID to LB creation.
+  - The annotation `service.beta.kubernetes.io/aws-load-balancer-managed-security-group` (new) msut be set to `true`, then creates the SG with required rules for ingress (based on listeners) and egress (based on service and health check ports).
+  - CCM LB controller manages the SG lifecycle (controllers may exists in CLB).
+
+  - The CCM's service controller will watch for Service creations and updates.
+  - When it encounters a Service with the annotation `service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true"` and `service.beta.kubernetes.io/aws-load-balancer-type: nlb`, the CCM will:
+    - Create a new AWS Security Group for the NLB. The name should follow a convention like `k8s-elb-a<generated-name-from-service-uid>`.
```

Comment:
> The name should follow a convention like `k8s-elb-a<generated-name-from-service-uid>

This is a interesting point, the convention for CCM to create NLB from Services is different than the ALBC, which follow the pattern: `k8s-<namespace>-<service_name>-<id>`

Reply:
Furthermore, I see NLB tags aren't standardized too:
CCM:
```
kubernetes.io/cluster/clusterID: owned
kubernetes.io/service-name: namespace/service-name
```

ALBC:
```
elbv2.k8s.aws/cluster: clusterID
service.k8s.aws/resource: LoadBalancer
service.k8s.aws/stack: namespace/service-name
```